cmd/kubeadm/app/cmd/config.go

```diff
@@ -138,6 +141,59 @@ func getDefaultAPIObjectBytes(apiObject string) ([]byte, error) {
 	return []byte{}, fmt.Errorf("--api-object needs to be one of %v", availableAPIObjects)
 }
 
+// NewCmdConfigMigrate returns cobra.Command for "kubeadm config migrate" command
+func NewCmdConfigMigrate(out io.Writer) *cobra.Command {
+	var cfgPath string
+	cmd := &cobra.Command{
+		Use:   "migrate",
+		Short: "Read an older version of the kubeadm configuration API types from a file, and output the similar config object for the newer version.",
+		Long: fmt.Sprintf(dedent.Dedent(`
+			This command lets you convert configuration objects of older versions to the latest supported version.
+			In this version of kubeadm, the following API versions are supported:
+			- %s
+			- %s
+			Further, kubeadm can only write out config of version %q, but read both types.
+			So regardless of what version you pass to the --config parameter here, the API object will be
+			read, deserialized, defaulted, converted, validated, and re-serialized when written to stdout.
+			In other words, the output of this command is what kubeadm actually would read internally if you
+			submitted this file to "kubeadm init"
+		`), kubeadmapiv1alpha2.SchemeGroupVersion.String(), kubeadmapiv1alpha1.SchemeGroupVersion.String(), kubeadmapiv1alpha2.SchemeGroupVersion.String()),
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(cfgPath) == 0 {
+				kubeadmutil.CheckErr(fmt.Errorf("The --config flag is mandatory"))
+			}
+
+			b, err := ioutil.ReadFile(cfgPath)
+			kubeadmutil.CheckErr(err)
+
+			var outputBytes []byte
+			gvk, err := kubeadmutil.GroupVersionKindFromBytes(b, kubeadmscheme.Codecs)
+			kubeadmutil.CheckErr(err)
+
+			switch gvk.Kind {
+			case masterConfig:
+				internalcfg, err := configutil.ConfigFileAndDefaultsToInternalConfig(cfgPath, &kubeadmapiv1alpha2.MasterConfiguration{})
+				kubeadmutil.CheckErr(err)
+
+				outputBytes, err = kubeadmutil.MarshalToYamlForCodecs(internalcfg, kubeadmapiv1alpha2.SchemeGroupVersion, kubeadmscheme.Codecs)
+				kubeadmutil.CheckErr(err)
+			case nodeConfig:
+				internalcfg, err := configutil.NodeConfigFileAndDefaultsToInternalConfig(cfgPath, &kubeadmapiv1alpha2.NodeConfiguration{})
+				kubeadmutil.CheckErr(err)
+
+				outputBytes, err = kubeadmutil.MarshalToYamlForCodecs(internalcfg, kubeadmapiv1alpha2.SchemeGroupVersion, kubeadmscheme.Codecs)
+				kubeadmutil.CheckErr(err)
+			default:
+				kubeadmutil.CheckErr(fmt.Errorf("Didn't recognize type with GroupVersionKind: %v", gvk))
+			}
+
+			fmt.Fprintf(out, string(outputBytes))
```

Comment:
I think we can provide another flag to write the content to a file. WDYT?

Reply:
Done